runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java

```diff
@@ -0,0 +1,156 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.stealingartefacts;
+
+import com.google.common.collect.ImmutableSet;
+import lombok.AccessLevel;
+import lombok.Getter;
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import net.runelite.api.ItemContainer;
+import net.runelite.api.ItemID;
+import net.runelite.api.Player;
+import net.runelite.api.Varbits;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.ItemContainerChanged;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.OverlayManager;
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+@PluginDescriptor(
+		name = "Stealing Artefacts",
+		description = "Show the location and status of your stealing artefact task.",
+		tags = {"thieving"},
+		enabledByDefault = false
+)
+public class StealingArtefactsPlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private StealingArtefactsOverlay overlay;
+
+	@Getter(AccessLevel.PACKAGE)
+	private StealingArtefactState stealingArtefactState;
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean isInPortPiscariliusRegion;
+
+	private static final ImmutableSet<Integer> PORT_PISCARILIUS_REGIONS = ImmutableSet.of(6970, 7226);
+
+	private static final ImmutableSet<Integer> ARTEFACT_ITEM_IDS = ImmutableSet.of(ItemID.STOLEN_PENDANT,
+			ItemID.STOLEN_GARNET_RING, ItemID.STOLEN_CIRCLET, ItemID.STOLEN_FAMILY_HEIRLOOM, ItemID.STOLEN_JEWELRY_BOX);
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		overlayManager.add(overlay);
+
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			clientThread.invoke(() ->
+			{
+				stealingArtefactState = StealingArtefactState.getStealingArtefactState(client.getVar(Varbits.STEALING_ARTEFACT_STATE));
+			});
+		}
+		else
+		{
+			stealingArtefactState = null;
+		}
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		overlayManager.remove(overlay);
+	}
+
+	private boolean hasArtefact()
+	{
+		ItemContainer inventory = client.getItemContainer(InventoryID.INVENTORY);
+
+		if (inventory != null)
+		{
+			final Stream<Item> items = Arrays.stream(inventory.getItems());
+			return items.anyMatch(item -> ARTEFACT_ITEM_IDS.contains(item.getId()));
+		}
+
+		return false;
+	}
+
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		int stealingArtefactStateVarbit = client.getVar(Varbits.STEALING_ARTEFACT_STATE);
+		StealingArtefactState state = StealingArtefactState.getStealingArtefactState(stealingArtefactStateVarbit);
```

Comment:
Since the varbit value itself is only ever used to generate a state, can we wrap this up into a `getCurrentStealingState` function to be used here and in `startUp()`?